upstream-sha256: 2413e18a9f86b84e2cd6ea4185a11aeb5e5ae4a31c7c7233bc6da47d87524580
actual-sha256: 3c088d7dff70a8d210ca56dcdb3450e0acdba4aaffbd7315b0a07d67d5324e4b
--- upstream.expected
+++ tolk.actual
@@ -1,26 +1,26 @@
 #include <metal_stdlib>
 using namespace metal;
-struct args_t { device float* data0_6; device unsigned char* data1_6; device float* data2_256; };
+struct args_t { device float* data0_6; device uchar* data1_6; device float* data2_256; };
 kernel void E_2_3(constant args_t& args [[buffer(0)]], uint3 gid [[threadgroup_position_in_grid]], uint3 lid [[thread_position_in_threadgroup]]) {
-  device float* data0_6 = args.data0_6; device unsigned char* data1_6 = args.data1_6; device float* data2_256 = args.data2_256;
+  device float* data0_6 = args.data0_6; device uchar* data1_6 = args.data1_6; device float* data2_256 = args.data2_256;
   int gidx0 = gid.x; /* 2 */
   int lidx0 = lid.x; /* 3 */
   int alu0 = (lidx0+(gidx0*3));
-  unsigned char val0 = (*(data1_6+alu0));
+  uchar val0 = (*(data1_6+alu0));
   float val1 = (*(data2_256+val0));
   *(data0_6+alu0) = val1;
 }
 ---
 #include <metal_stdlib>
 using namespace metal;
-struct args_t { device float* data0_192; device unsigned char* data1_96; device float* data2_16; device float* data3_6; };
+struct args_t { device float* data0_192; device uchar* data1_96; device float* data2_16; device float* data3_6; };
 kernel void E_3_32_2(constant args_t& args [[buffer(0)]], uint3 gid [[threadgroup_position_in_grid]], uint3 lid [[thread_position_in_threadgroup]]) {
-  device float* data0_192 = args.data0_192; device unsigned char* data1_96 = args.data1_96; device float* data2_16 = args.data2_16; device float* data3_6 = args.data3_6;
+  device float* data0_192 = args.data0_192; device uchar* data1_96 = args.data1_96; device float* data2_16 = args.data2_16; device float* data3_6 = args.data3_6;
   int gidx0 = gid.x; /* 3 */
   int lidx0 = lid.x; /* 32 */
-  unsigned char val0 = (*(data1_96+(lidx0+(gidx0<<5))));
-  float val1 = (*(data2_16+(val0>>((unsigned char)(4u)))));
-  float val2 = (*(data2_16+(val0&((unsigned char)(15u)))));
+  uchar val0 = (*(data1_96+(lidx0+(gidx0<<5))));
+  float val1 = (*(data2_16+(val0>>((uchar)(4u)))));
+  float val2 = (*(data2_16+(val0&((uchar)(15u)))));
   float val3 = (*(data3_6+((gidx0<<1)+(lidx0>>4))));
   *((device float2*)((data0_192+((gidx0<<6)+(lidx0<<1))))) = float2((val2*val3),(val1*val3));
 }
